src/main/java/app/model/agents/DirectionFollowAgent/GoToTargetAgent.java

```diff
@@ -0,0 +1,94 @@
+package app.model.agents.DirectionFollowAgent;
+
+import app.controller.graphicsEngine.Ray;
+import app.controller.linAlg.Angle;
+import app.controller.linAlg.Vector;
+import app.model.Move;
+import app.model.Type;
+import app.model.agents.Agent;
+import app.model.agents.AgentImp;
+import app.model.agents.World;
+
+public class GoToTargetAgent extends AgentImp
+{
+    private Ray targetRay;
+    private boolean guardDetected;
+    private final boolean DEBUG = false;
+
+    public GoToTargetAgent(Vector position, Vector direction, double radius, Type type, Ray targetRay, World world)
+    {
+        super(position, direction, radius, type);
+        this.targetRay = targetRay;
+        this.world = world;
+        guardDetected = false;
+    }
+
+    public Move move()
+    {
+        return goToTarget();
+    }
+
+    private Move goToTarget()
+    {
+        if(!direction.equals(targetRay.direction()))
+        {
+            return new Move(targetRay.direction(), new Vector(0, 0));
+        } else
+        {
+            return new Move(targetRay.direction(), targetRay.direction().scale(moveLength));
+        }
+    }
+
+    public boolean guardDetected()
+    {
+        for(Ray r : view)
+        {
+            if(r.getType()==Type.GUARD)
+            {
+                if(DEBUG) {System.out.println("     GUARD DETECTED!");}
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Vector closestGuard()
```

Comment:
This method may be better suited within the `AgentImp` class, as all implementations will need to be able to detect other agents

Reply:
good idea, I'll move it